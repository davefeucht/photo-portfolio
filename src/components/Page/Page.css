.page {
  min-height: 100%;
  display: grid;
  grid-template-columns: 100%;
  grid-template-areas: "page-header" "page-content";
  grid-template-rows: .25fr 4fr;
}