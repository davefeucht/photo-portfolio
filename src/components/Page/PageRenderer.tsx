import PageContent from 'components/PageContent/PageContent';
import SectionHeader from 'components/SectionHeader/SectionHeader';
import * as React from 'react';

interface Props {
    title: string;
    content: string;
}

const PageRenderer: React.FC<Props> = ({
    title,
    content
}) => {
    return (
        <div className="page">
            <SectionHeader title={title} />
            <PageContent content={content} />
        </div>
    );
};

export default PageRenderer;
