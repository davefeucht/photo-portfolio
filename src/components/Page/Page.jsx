/** ***************
* Page component implements displaying a page in the application
**************** */

import './Page.css';

import { observer } from 'mobx-react';
import PropTypes from 'prop-types';
import React, { useEffect } from 'react';
import { useParams } from 'react-router-dom';

import { getPage } from '../../utils/Api.js';
import PageContent from '../PageContent/PageContent.jsx';
import SectionHeader from '../SectionHeader/SectionHeader.jsx';

function Page({ stateStore }) {
    const { pageId } = useParams();
    const { currentPageData } = stateStore;

    useEffect(() => {
        getPage(pageId, stateStore.siteInfo.siteUrl)
            .then(page => {
                stateStore.setPageData(page);
            });
    }, [pageId]);

    return (
        <div className="page">
            <SectionHeader title={currentPageData.title?.rendered} />
            <PageContent content={currentPageData.content?.rendered} />
        </div>
    );
}

Page.displayName = 'Page';

Page.propTypes = {
    stateStore: PropTypes.object.isRequired
};

export default observer(Page);
