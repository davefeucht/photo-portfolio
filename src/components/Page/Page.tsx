/** ***************
* Page component implements displaying a page in the application
**************** */

import './Page.css';

import { observer } from 'mobx-react';
import * as React from 'react';
import { useEffect } from 'react';
import { useParams } from 'react-router-dom';

import { getPage } from '../../utils/Api';
import {
    Page as PageState,
    SiteInfo
} from '../../utils/types';
import PageContent from '../PageContent/PageContent';
import SectionHeader from '../SectionHeader/SectionHeader';

interface PageProps {
    currentPageData: PageState,
    siteInfo: SiteInfo,
    setPageData: (a: PageState) => void
}

const Page: React.FC<PageProps> = ({ currentPageData, siteInfo, setPageData }) => {
    const { pageId } = useParams();

    useEffect(() => {
        if (pageId) {
            getPage(parseInt(pageId), siteInfo.siteUrl)
                .then(page => {
                    setPageData(page);
                });
        }
    }, [pageId]);

    return (
        <div className="page">
            <SectionHeader title={currentPageData.title?.rendered} />
            <PageContent content={currentPageData.content?.rendered} />
        </div>
    );
};

Page.displayName = 'Page';

export default observer(Page);
