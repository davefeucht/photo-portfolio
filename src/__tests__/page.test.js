import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Page from '../components/Page/Page';
import stateStore from '../StateStore/store';

const page = {
    id: 150,
    content: {
        rendered: 'Contact me!'
    },
    title: {
        rendered: 'Contact'
    }
};

jest.mock('../utils/Api', () => ({
    getPage: () => Promise.resolve(page)
}));

test('Page displays', () => {
  const store = new stateStore();
  let component;
  let tree;
  component = renderer.create(
      <MemoryRouter initialEntries={['/page/150']} >
          <Page stateStore={store}></Page>
      </MemoryRouter>
  );
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
