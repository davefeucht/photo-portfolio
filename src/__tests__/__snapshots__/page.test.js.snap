// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category displays 1`] = `
<div
  className="page"
>
  <div
    className="section-header"
  >
    <div
      className="title"
    />
  </div>
  <div
    className="page__content"
    dangerouslySetInnerHTML={
      {
        "__html": undefined,
      }
    }
  />
</div>
`;

exports[`Page displays 1`] = `
<div
  className="page"
>
  <div
    className="section-header"
  >
    <div
      className="title"
    />
  </div>
  <div
    className="page__content"
    dangerouslySetInnerHTML={
      {
        "__html": undefined,
      }
    }
  />
</div>
`;
